### PR TITLE
[snapshots] QEMU-specific snapshot taking

### PR DESCRIPTION
**Pull request by ricab**

### CODE REVIEW

**Review by sharder996 — COMMENTED** [review]
Looks good, just a few nitpicks to make function signatures consistent.

Also, builds are failing on macOS so I'm unable to test there atm

File: src/platform/backends/qemu/qemu_snapshot.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+} // namespace
+
+mp::QemuSnapshot::QemuSnapshot(const std::string& name, const std::string& comment,
+                               std::shared_ptr<const Snapshot> parent, const VMSpecs& specs, const QString& image_path)
+    : BaseSnapshot(name, comment, std::move(parent), specs), image_path{image_path}
+{
+}
+
+mp::QemuSnapshot::QemuSnapshot(const QJsonObject& json, const multipass::QemuVirtualMachine& vm)
```
[the remark this answers is not in the document]
Comment (ricab) [reply src/platform/backends/qemu/qemu_snapshot.cpp]: Yep. CLion doing this to me...

**Comment by ricab** [comment]
Thanks for noticing @sharder996. I have addressed your requests.

**Review by sharder996 — APPROVED** [review]
Tested on M2 Mac and Linux and appears to be functioning correctly! Verified snapshots are being taken and all snapshot related files are being removed upon instance deletion.

**Comment by ricab** [comment]
Cool, thanks a lot @sharder996!

**Pull request merged**